cmake/FindDPCPP.cmake

```
@@ -38,6 +38,9 @@ else()
 endif()
 set(CMAKE_CXX_FLAGS "${DPCPP_FP_FLAG} ${CMAKE_CXX_FLAGS}")
 
+# Disable range rounding feature to reduce # of SYCL kernels.
+set(CMAKE_CXX_FLAGS "-D__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__")
```

```suggestion
set(CMAKE_CXX_FLAGS "-fsycl-disable-range-rounding")
```

---

I tried this but got an `unknown argument` error!

---

Thanks. I mart this as resolved and take a look at this error.

---

This patch resets C++ compiler flags with just one define. It should be `set(CMAKE_CXX_FLAGS "..." $CMAKE_CXX_FLAGS)` to keep previously set flags or better use specific CMake command:

```suggestion
add_compile_definitions(__SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING__)
```

I suppose that DPC++ now builds SYCL sources in C++ mode, not SYCL mode (i.e. w/o `-fsycl` flag).

I looked at `-fsycl-disable-range-rounding` issue and I see that this is front-end option, so we can pass it via -Xclang. On the other hand, there is no additional value in using this option - it just sets the macro, so let's set the macro directly instead.

---

You are right of course, although I did copy that commit from [somewhere else](https://github.com/KhronosGroup/SYCL-CTS/pull/234/commits/d752a191409a48449da55e0a86fe768bf3c559d5) ;-). Will fix.

---

I went for `CMAKE_CXX_FLAGS`  for consistency with the surrounding code.

---

> You are right of course, although I did copy that commit from [somewhere else](https://github.com/KhronosGroup/SYCL-CTS/pull/234/commits/d752a191409a48449da55e0a86fe768bf3c559d5) ;-). Will fix.

Watch out for this guy - as you see he pushes buggy code to other's pull requests. :)